### ISSUE TITLE
wl-present does not exit after wl-mirror is closed

### ISSUE THREAD

**Issue by Vladimir-csp** [Ferdi265/wl-mirror#36]
It keeps pipectl process and itself active if any other command was given.

```
wl-present mirror
# select output
# close wl-mirror window
# wl-present exits
```

```
wl-present mirror &
# select output
wl-present set-region
# select region
# close wl-mirror window
jobs
[1]+  Running                 wl-present mirror &
```

**Comment by Ferdi265** [comment]
Does this issue also occur with `WL_PRESENT_PIPECTL=pipectl-shim` or only with real `pipectl`?

I have definitely seen this before, but haven't had the energy to fully debug this yet. That part of wl-present has always been a bit hacky.

Thanks for reporting, I'll get to it once I find a bit more time for wl-mirror again.

**Comment by Vladimir-csp** [comment]
It does not happen with pipectl-shim. 

Would you be interested in me working on wl-present a bit? I see some potential in optimizing and improving it. For example I do not see any really necessary bash-specific features used, so I've converted it to pure shell.

I could try to add some concurrency safeguards, child process cleanups, etc.

**Comment by Vladimir-csp** [comment]
[pipectl problem](https://github.com/Ferdi265/pipectl/issues/1)

**Comment by Ferdi265** [comment]
This will be fixed properly once pipectl 0.5.0 is released. The fix is already in pipectl-git.

**Comment by Ferdi265** [comment]
Fixed upstream in [pipectl 0.5.0](https://github.com/Ferdi265/pipectl/releases/tag/v0.5.0)
